packages/dds/tree/src/feature-libraries/default-schema/schemaChecker.ts

```diff
@@ -50,38 +60,38 @@ export function isNodeInSchema(
 		if (!allowsValue(schema.leafValue, node.value)) {
 			return SchemaValidationErrors.LeafNode_InvalidValue;
 		}
-	} else if (schema instanceof ObjectNodeStoredSchema) {
+	} else {
```

Comment:
The changes here just deduplicate the logic for rejecting nodes with values by pulling it out of the two else if branches to here.